### PR TITLE
MLX provider.

### PR DESCRIPTION
**Pull request by alexander-sannikov**
Implemented: FI_TAGGED

### CODE REVIEW

File: prov/mlx/src/mlx_av.c [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+	av = container_of(fi_av, struct mlx_av, av);
+	if ((av->async) && (!av->eq))
+	{
+		return -FI_ENOEQ;
+	}
+
+	for (i = 0; i < count; ++i)
+	{
+		ucp_ep_destroy((ucp_ep_h)(fi_addr[i]));
+	}
```
Comment (shefty) [inline prov/mlx/src/mlx_av.c]: nit: here and in other places, please follow the Linux kernel coding conventions.  I.e. put the open brace at the end of the if/for/while lines.

File: prov/mlx/src/mlx_ep.c [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+static int mlx_ep_bind(struct fid *fid, struct fid *bfid, uint64_t flags)
+{
+	struct mlx_ep *ep;
+	struct util_cq *cq;
+
+	ep = container_of(fid, struct mlx_ep, ep.ep_fid.fid);
+	int status = FI_SUCCESS;
+
+	switch (bfid->fclass) {
+		case FI_CLASS_CQ:
```
Comment (shefty) [inline prov/mlx/src/mlx_ep.c]: nit: align case with switch (tab in to same position)

**Comment by shefty** [comment]
Looks good - I only have some nit comments regarding coding style changes.

**Comment by alexander-sannikov** [comment]
@shefty code style issues are fixed.

**Pull request merged**